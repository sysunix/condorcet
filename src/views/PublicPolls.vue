<template>
  <Loader>
    <Empty
      :condition="publicPolls.length === 0"
      :text="$t('empty.publicPolls.text')"
      :action-text="$t('empty.publicPolls.action')"
      to="/polls/new"
    />

    <div class="flex flex-wrap">
      <div
        v-for="poll in publicPolls"
        :key="poll.id"
        class="w-full md:w-1/2 lg:w-1/3"
      >
        <Card
          class="md:mx-2"
          v-bind="poll"
          :user-id="userId"
          :is-joinable="true"
          @onJoin="joinPoll"
        />
      </div>
    </div>
  </Loader>
</template>

<script>
import { mapGetters, mapActions } from "vuex";
import Card from "../components/Card";
import Loader from "../components/Loader";
import Empty from "../components/Empty";

export default {
  name: "PublicPolls",
  components: {
    Card,
    Loader,
    Empty
  },
  computed: {
    ...mapGetters("user", ["userId"]),
    ...mapGetters("poll", ["publicPolls"])
  },
  created() {
    this.fetchPublicPolls();
  },
  methods: {
    ...mapActions("poll", ["fetchPublicPolls"]),
    joinPoll(id) {
      this.$router.push({
        name: "poll_join",
        params: { id },
        query: { token: this.publicPolls.find(poll => poll.id === id).token }
      });
    }
  }
};
</script>

<style></style>
