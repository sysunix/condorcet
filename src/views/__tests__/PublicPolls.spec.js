import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import PublicPolls from "../PublicPolls.vue";
import {
  MOCK_POLL,
  MOCK_POLL_ID,
  MOCK_USER_ID,
  getTranslation
} from "../../utils/test";

const localVue = createLocalVue();
localVue.use(Vuex);

jest.mock("../../firebase");

const mocks = {
  $router: { push: jest.fn() },
  $t: getTranslation
};

describe("PublicPolls.vue", () => {
  let wrapper;

  beforeEach(() => {
    const store = new Vuex.Store({
      modules: {
        user: {
          namespaced: true,
          getters: {
            userId: () => MOCK_USER_ID
          }
        },
        poll: {
          namespaced: true,
          actions: {
            fetchPublicPolls: () => {}
          },
          getters: {
            publicPolls: () => [{ ...MOCK_POLL, id: MOCK_POLL_ID }]
          }
        }
      }
    });

    wrapper = shallowMount(PublicPolls, { localVue, store, mocks });
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should join poll", () => {
    wrapper.vm.joinPoll(MOCK_POLL_ID);
    expect(wrapper.vm.$router.push).toBeCalled();
  });
});
