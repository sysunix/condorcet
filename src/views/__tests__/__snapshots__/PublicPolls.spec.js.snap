// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicPolls.vue should render 1`] = `
<loader-stub>
  <empty-stub text="Il n'y a pas de scrutins publics disponibles." actiontext="CrÃ©er un scrutin" to="/polls/new"></empty-stub>
  <div class="flex flex-wrap">
    <div class="w-full md:w-1/2 lg:w-1/3 py-2">
      <card-stub id="MOCK_POLL_ID" userid="MOCK_USER_ID" description="Qui repellendus est aliquam vel qui." question="Qui est le meilleur ?" owner="MOCK_USER_ID" isactive="true" users="MOCK_USER_ID" is-joinable="true" answers="[object Object],[object Object],[object Object]" ispublic="true" timestamp="[object Object]" token="MOCK_TOKEN" class="md:mx-2"></card-stub>
    </div>
  </div>
</loader-stub>
`;
